framework/src/modules/chain/bft/headers_list.js

```diff
@@ -103,6 +110,21 @@ class HeadersList {
 		);
 	}
 
+	top(size) {
+		assert(size, 'Please provide the size');
+
+		return this.items.slice(this.length - size, this.length + 1);
+	}
+
+	getBlockHeaderForDelegate(publicKey, { fromTop } = {}) {
```

Comment:
Other method are very generic but this is very specific.
Maybe put this logic on the user side

Reply:
I will tend to make it more generic then.

Reply:
Removed this method and used specific inside bft.